### ISSUE TITLE
Add Support for 2FA

### ISSUE THREAD

**Issue by Locutus73** [homebridge-plugins/homebridge-arlo#35]
Arlo just introduced 2 factor authentication.
Old authentication API still works for accounts with 2FA disabled, but enabling 2FA would be a great security improvement.

Thank you in advance.
Regards.

**Comment by donavanbecker** [comment]
pull requests are welcome.

**Comment by Phoenix-DH** [comment]
Are there any news concerning 2FA?
Is there a chance to integrate it?

**Comment by 1simonfish** [comment]
What would the user experience be for using 2FA? Would the user authenticate only when Homebridge (re)starts? How would the user know they need to input the code?

**Comment by Phoenix-DH** [comment]
Have absolutely no clue...
I was still not forced to enable it.

**Comment by OxfordRancher** [comment]
Bump please The last time I logged into Arlo (occasionally I get logged out of the app) there was a notice that 2FA would be mandatory by 31 Oct 2020.

It seems native support (from Arlo) is only for new models since I have the older wire free camera with CR123 batteries I would like to get them working via this plugin.

Also I already have ffmpeg configured for Kuna (Maximus light) would there be a conflict with ffmpeg? The Kuna doesn’t stream at present it just pushes the static snapshot , which serves 80% of my need.

I also have a Wyze pan cam but have not flashed it with the rtsp firmware/ configured it but I assume it may also rely on ffmpeg. In a perfect world all of my cameras would be supported by ffmpeg either directly or with a native plugin making calls to ffmpeg (I would settle for functionality at least at the snapshot level ).

**Comment by Neutral-President** [comment]
I’m not even all that interested in getting camera output through Homebridge. I mainly want to be able to have HomeKit (via Shortcuts, Automations, and Siri) to be able to easily change modes without having to go use the Arlo app.

**Comment by peatoe** [comment]
2FA is mandatory by the end of the day today. I also have an old Arlo Wire-Free camera that takes CR123 Batteries (The ones that say Netgear on the side of the camera). As of right now my Arlo Hub is the only device showing in Homekit. I try to add my two cameras but Homekit can't see them. It use to but now after resetting it up they don't show

**Comment by charlesvestal** [comment]
> What would the user experience be for using 2FA? Would the user authenticate only when Homebridge (re)starts? How would the user know they need to input the code?

Fore reference, pyaarlo appears to be implementing this based on logging into the email to which the code is sent (or an app that does the same) https://github.com/twrecked/pyaarlo#2fa

**Comment by BjornB2** [comment]
Installed the plugin just now and found that it wasn't working and was already suspecting 2FA.
Just wondering: is 2FA on the roadmap? I am not in a hurry or something, but if it's not going to be added I can better remove the plugin obviously 👍

**Comment by OxfordRancher** [comment]
I also have older wire free Arlo cameras, I can’t see my base station or cameras. I am able to log in via app and web and do not have 2Fa

**Comment by bilcastine** [comment]
I'm not a developer, but wanted to vote this up, I am also seeing this issue since MFA is mandatory.

**Comment by bilcastine** [comment]
> What would the user experience be for using 2FA? Would the user authenticate only when Homebridge (re)starts? How would the user know they need to input the code?

as with other systems implementing MFA, I would expect successful authentication (username, password, MFA) to issue a token for future authentication to Arlo, so only one-time setup would be required. some services (G Suite for example) allow administrators to set a token expiration timeout, after which the user would need to manually log in again to get a new auth token, though I'd be surprised if a consumer product provided that level of customization.

**Comment by jonnyborbs** [comment]
Upvote for this as well, it appears this plugin is now entirely nonfunctional since the 2FA requirement went into effect. I was hoping to use it to stream video from my wired Q cameras that don't natively support HomeKit, but there's no accessories returned at all when the platform initializes, and I presume it's due to the lack of 2FA support.

**Comment by sanstarr** [comment]
is there any chance to implement the 2fa feature? :/
plugin is not useful at the moment.

thank you!!!

**Comment by gminors** [comment]
Yes please this app update would be very useful to get it working again.
